From 5ca9c15acd175fcf6a191f22ea49456cde49d34f Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Thu, 26 Mar 2020 01:54:35 +0000
Subject: [PATCH] fix mailboxes with tildes

Config variables that contain filesystem paths, should have `~` expanded
before saving.  For `DT_PATH`, this is done automatically.

Restore `reset_tilde()` so that `DT_MAILBOX` get their initial values correct.

Fixes: #2206
---
 main.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/main.c b/main.c
index 31c218b36..3d82dd33f 100644
--- a/main.c
+++ b/main.c
@@ -107,6 +107,29 @@ typedef uint8_t CliFlags;         ///< Flags for command line options, e.g. #MUT
 #endif
 // clang-format on
 
+/**
+ * reset_tilde - Temporary measure
+ * @param cs Config Set
+ */
+static void reset_tilde(struct ConfigSet *cs)
+{
+  static const char *names[] = { "folder", "mbox", "postponed", "record" };
+
+  struct Buffer value = mutt_buffer_make(256);
+  for (size_t i = 0; i < mutt_array_size(names); i++)
+  {
+    struct HashElem *he = cs_get_elem(cs, names[i]);
+    if (!he)
+      continue;
+    mutt_buffer_reset(&value);
+    cs_he_initial_get(cs, he, &value);
+    mutt_buffer_expand_path_regex(&value, false);
+    cs_he_initial_set(cs, he, value.data, NULL);
+    cs_he_reset(cs, he, NULL);
+  }
+  mutt_buffer_dealloc(&value);
+}
+
 /**
  * mutt_exit - Leave NeoMutt NOW
  * @param code Value to return to the calling environment
@@ -535,6 +558,8 @@ int main(int argc, char *argv[], char *envp[])
   }
 #endif
 
+  reset_tilde(cs);
+
   if (dfile)
   {
     cs_str_initial_set(cs, "debug_file", dfile, NULL);
